addon/authenticators/devise.js

```diff
@@ -81,6 +81,8 @@ export default BaseAuthenticator.extend({
     if (!isEmpty(tokenAttribute) && !isEmpty(identificationAttribute)) {
       return Promise.resolve(data);
     } else {
+      this._validate(Promise.reject, data, tokenAttributeName);
+      this._validate(Promise.reject, data, identificationAttributeName);
       return Promise.reject();
```

Comment:
Might be easier to return a boolean from `_validate` and get `tokenAttributeName` and `identificationAttributeName` internally in the method so that you could just do

```js
if (!this._validate(data)) {
  return Promise.reject();
}
```